src/pages/setup/SiteSetupForm.jsx

```diff
@@ -39,12 +39,12 @@ export default function SiteSettings({ primaryButtonId }) {
 
   const [currentValues, setCurrentValues] = useState(null);
 
-  const edmValues = newSettingFields.map(fieldKey =>
-    get(newSiteSettings, ['data', fieldKey, 'value']),
-  );
   useEffect(() => {
+    const edmValues = newSettingFields.map(fieldKey =>
+      get(newSiteSettings, ['data', fieldKey, 'value']),
+    );
     setCurrentValues(zipObject(newSettingFields, edmValues));
-  }, edmValues);
+  }, [newSettingFields]);
```

Comment:
This `useEffect` now has a dependency on `newSiteSettings`.
Also, `newSettingFields` does not need to be a dependency because it is defined outside of the component.